tests/test_codelist.py

```diff
@@ -80,3 +80,13 @@ def test_to_excel(tmpdir):
     exp = pd.read_excel(TEST_DATA_DIR / "validation_nc.xlsx")
 
     pdt.assert_frame_equal(obs, exp)
+
+
+def test_stray_tag_fails():
+    """Check that typos in a tag raises expected error"""
+
+    match = "Unexpected {} in codelist : Primary Energy|{Feul}"
```

Comment:
why?

Reply:
Ah! Ok sorry got it. I thought the `match` looks for this string in the error, and it does not matter if there are more characters. but maybe not... Should I change it?

Reply:
OK no you're right, something is wrong, thank you! I'll look into that

Reply:
Just played around for a minute, it seems that the `:` does something to the match-validation.

I suggest to remove the whitespace, so that the test passes as expected (by failing with the right error message).
```suggestion
    match = "Unexpected {} in codelist: Primary Energy|{Feul}"
```

And I suggest that you look into this more closely tomorrow (but not related to this PR), for your own knowledge...

Reply:
> Ah! Ok sorry got it. I thought the match looks for this string in the error, and it does not matter if there are more characters.

As far as my understanding goes that is correct. I think the check works as follows (for non-regex):
```python
match in str(error)
```
which in our case since @luciecastella just added additional information to the error should still evaluate to true.

Reply:
The match is interpreted as a regular expression and the character messing up the match is the pipe `|` which is evaluated as _or_. To evaluate it as a literal pipe character you'll have to prefix it with a slash `\`.
```python
match = "Unexpected {} in codelist : Primary Energy\|{Feul}"
```
should fail as expected and
```suggestion
    match = "Unexpected {} in codelist: Primary Energy\|{Feul}"
```
should be evaluated correctly and make the test pass.

Reply:
Great thank you!!

Reply:
Indeed, the pipe in the string allows for any string to pass the test. We actually need a double `\\` to make it work :)